### PR TITLE
Refactor upload logic (towards staged releases)

### PR DESCRIPTION
**Pull request by carlopi**
Various different changes connected to https://github.com/duckdb/duckdb/pull/11133.

Idea is moving workflows so that they can be invoked via workflow_call and passing them a regular set of parameters.
This would allow long term more composable workflows, and short term to have a scheduled jobs preparing staged releases artifact ahead of time.

To keep changes manageable, rework is currently done only for workflows that do generate relevant artifacts, basically whatever was uploaded via `scripts/asset-upload-gha.py`.

One of the bonuses of this rework:
* skip_tests can be provided to workflows, and it's not anymore tied to a tag being present
Decoupling this logic allows both more flexibility AND allows to reduce codepaths tested only on release day

Adding `if: ${{ skip_tests == 'true' }}` can (and should) be considered even in more places, but this can also be an incremental change. I added a global variable allowing skipping all Python tests, that can come handy given tests might be finicky in different ways, and we might want to do without the randomness during release processes.

Not done as part of this PR but yet to come:
* fix Python libraries to inherit proper version from OVERRIDE_GIT_DESCRIBE
* move out of `twine upload` and into staged releases also for Python
* add (and test) the script invoking `python3 scripts/asset-upload-gha.py` on the staged artifcats

This PR has been tested in a fork of duckdb with the right privileges, more stress testing underway, but I feel this chunk of work can be already considered for review and eventually merging.

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
Looks great, thanks!

File: .github/workflows/Python.yml [hunk 1]
```diff
@@ -114,8 +133,8 @@ jobs:
         s3_id: ${{ secrets.S3_ID }}
         s3_key: ${{ secrets.S3_KEY }}
         signing_pk: ${{ secrets.DUCKDB_EXTENSION_SIGNING_PK }}
-        run_tests: ${{ startsWith(github.ref, 'refs/tags/v') && 0 || 1  }}
-        run_autoload_tests: ${{ startsWith(github.ref, 'refs/tags/v') && 0 || 1  }}
+        run_tests: ${{ inputs.skip_tests == 'true' && 0 || 1  }}
```
Comment (samansmink) [inline .github/workflows/Python.yml:136]: this ternary like thing is broken I think, this is actually part of what caused the whole 0.10.0 release issue. It only works if the truthy value evaluates to true: https://github.com/actions/runner/issues/409#issuecomment-752775072. Inverting the condition should fix it

**Review by samansmink — COMMENTED** [review]
Looks great! love the composability of this

File: .github/workflows/Python.yml (hunk 1 above)
Comment (carlopi) [reply .github/workflows/Python.yml:136]: just to double check, should I change it to:
```
run_tests: ${{ inputs.skip_tests != 'true' && 1 || 0  }}
```
?

Comment (samansmink) [reply .github/workflows/Python.yml:136]: jep I think that should do it

**Comment by carlopi** [comment]
Currently there is a CI failure due to networking, but that's OK, but then on the private fork where I am testing this I found another couple of minor problems, I will update the PR in a bit also with the fix by @samansmink.

File: .github/workflows/Python.yml (hunk 1 above)
Comment (carlopi) [reply .github/workflows/Python.yml]: Should be fixed by: https://github.com/duckdb/duckdb/pull/11156/commits/127b5ef5b794ca09fab5aebfed5a50bccd20f9ff

**Comment by carlopi** [comment]
Added some minor fixes, I am not sure about syntax of the Python upload where before we were using this syntax:
```
python scripts/asset-upload-gha.py duckdb_python_src.tar.gz=tools/pythonpkg/dist/duckdb-*.tar.gz
```
and now file has to be explicitly provided. I think I got it but it needs a while to test it properly.

**Pull request merged**